app/app.go

```diff
@@ -9,276 +9,255 @@ import (
 	dbm "github.com/tendermint/tm-db"
-
-	abci "github.com/tendermint/tendermint/abci/types"
-	tmos "github.com/tendermint/tendermint/libs/os"
-
-	"github.com/cosmos/cosmos-sdk/version"
-	distr "github.com/cosmos/cosmos-sdk/x/distribution"
-	"github.com/cosmos/cosmos-sdk/x/slashing"
-	"github.com/cosmos/cosmos-sdk/x/supply"
 )
 
 const (
 	appName = "akash"
 )
 
 var (
-	mbasics = module.NewBasicManager(
-		genutil.AppModuleBasic{},
 
-		// accounts, fees.
+	// ModuleBasics defines the module BasicManager is in charge of setting up basic,
+	// non-dependant module elements, such as codec registration
+	// and genesis verification.
+	ModuleBasics = module.NewBasicManager(
 		auth.AppModuleBasic{},
-
-		// tokens, token balance.
-		bank.AppModuleBasic{},
-
-		// total supply of the chain
 		supply.AppModuleBasic{},
-
-		// inflation
-		mint.AppModuleBasic{},
-
+		genutil.AppModuleBasic{},
+		bank.AppModuleBasic{},
 		staking.AppModuleBasic{},
-
-		slashing.AppModuleBasic{},
-
+		mint.AppModuleBasic{},
 		distr.AppModuleBasic{},
-
+		gov.NewAppModuleBasic(
+			paramsclient.ProposalHandler, distr.ProposalHandler,
+		),
 		params.AppModuleBasic{},
+		crisis.AppModuleBasic{},
+		slashing.AppModuleBasic{},
+		evidence.AppModuleBasic{},
 
 		// akash
 		deployment.AppModuleBasic{},
 		market.AppModuleBasic{},
 		provider.AppModuleBasic{},
 	)
+
+	// module account permissions
+	maccPerms = map[string][]string{
+		auth.FeeCollectorName:     nil,
+		distr.ModuleName:          nil,
+		mint.ModuleName:           {supply.Minter},
+		staking.BondedPoolName:    {supply.Burner, supply.Staking},
+		staking.NotBondedPoolName: {supply.Burner, supply.Staking},
+		gov.ModuleName:            {supply.Burner},
+	}
+
+	// module accounts that are allowed to receive tokens
+	allowedReceivingModAcc = map[string]bool{
+		distr.ModuleName: true,
+	}
 )
 
 // AkashApp extends ABCI appplication
 type AkashApp struct {
 	*bam.BaseApp
 	cdc *codec.Codec
 
+	invCheckPeriod uint
+
+	// keys to access the substores
 	keys  map[string]*sdk.KVStoreKey
 	tkeys map[string]*sdk.TransientStoreKey
 
-	keeper struct {
-		acct       auth.AccountKeeper
-		bank       bank.Keeper
-		params     params.Keeper
-		supply     supply.Keeper
-		staking    staking.Keeper
-		distr      distr.Keeper
-		slashing   slashing.Keeper
-		mint       mint.Keeper
-		deployment deployment.Keeper
-		market     market.Keeper
-		provider   provider.Keeper
-	}
-
+	// subspaces
+	subspaces map[string]params.Subspace
+
+	// keepers
+	AccountKeeper    auth.AccountKeeper
+	BankKeeper       bank.Keeper
+	SupplyKeeper     supply.Keeper
+	StakingKeeper    staking.Keeper
+	SlashingKeeper   slashing.Keeper
+	MintKeeper       mint.Keeper
+	DistrKeeper      distr.Keeper
+	GovKeeper        gov.Keeper
+	CrisisKeeper     crisis.Keeper
+	ParamsKeeper     params.Keeper
+	EvidenceKeeper   evidence.Keeper
+	DeploymentKeeper deployment.Keeper
+	MarketKeeper     market.Keeper
+	ProviderKeeper   provider.Keeper
```

Comment:
Why are these public?

Reply:
making the keepers public leverages the `AkashApp` as the source for integration tests (as it should be). See https://github.com/cosmos/cosmos-sdk/issues/4875 for reference

Reply:
fwiw I think that the reality is that they don't need to be fields in the struct at all

Reply:
You need to access all of the keepers?

Reply:
anyways, please keep them in a separate struct for organization/isolation and only export the keepers that you need to.

Reply:
> fwiw I think that the reality is that they don't need to be fields in the struct at all?

As you prefer, I'd still need to make the `keeper` internal struct public tho.

> You need to access all of the keepers?

I need access to all the keepers used by the Akash modules

Reply:
ok please keep them in an isolated struct and export them on an as-needed basis.